src/datepicker/index.es

```diff
@@ -0,0 +1,189 @@
+  }
+
+  getPosition() {
+    const { container } = this.els;
+
+    const { top } = screenPosition(container);
+    const { height } = size(container);
+    const { height: screenHeight } = screenSize(global);
+
+    const availableSpace = screenHeight - top;
+    const neededSpace = height * 2;
+    const isTop = neededSpace > availableSpace;
+
+    return { isTop };
+  }
+
+  activate() {
+    if (this.isVisible()) return;
+
+    this.stopListeningToKeys = keymanager('esc', this.deactivate);
+    this.stopListeningToClicks = eventproxy('click', this.handleGlobalClick);
+
+    const updater = () => {
+      const { isTop } = this.getPosition();
+      return { isVisible: true, isTop };
+    };
+
+    this.setState(updater);
+  }
+
+  deactivate() {
+    if (!this.isVisible()) return;
+
+    this.stopListeningToKeys();
+    this.stopListeningToClicks();
+
+    this.setState({ isVisible: false, isTop: false }, this.validate);
+  }
+
+  handleSelect(value) {
+    this.setValue(value, this.validate);
+    this.deactivate();
+  }
+
+  handleClick() {
+    if (!this.isVisible()) this.activate();
+    else this.deactivate();
+  }
+
+  handleClear() {
+    this.setValue(null);
+  }
+
+  isVisible() {
+    return this.state.isVisible;
+  }
+
+  isClearable() {
+    const { value } = this.state;
+    return value !== null && value !== undefined;
+  }
+
+  render() {
+    const className = clsx('c-datepicker', this.props.className);
+    const { value, isTop } = this.state;
+    const {
+      label,
+      placeholder,
+      children,
+      minDate,
+      maxDate,
+      dateFormat,
+      firstDay,
+      weekdayShortLabels,
+      monthLabels,
+    } = this.props;
+
+    const localizedValue = getValue(value, dateFormat);
+    const inputClassName = clsx('ui-input', { 'ui-input-error': this.isErrorActive() });
+
+    let labelNode;
+    if (label) labelNode = <strong className="ui-label">{label}</strong>;
+
+    let error;
+    if (this.isErrorActive()) error = <em className="ui-error">{this.getError()}</em>;
+
+    let picker;
+    if (this.isVisible()) {
+      picker = (
+        <Picker
+          className={clsx({ 'is-top': isTop })}
+          onChange={this.handleSelect}
+          selected={value}
+          {...{ theme, minDate, maxDate, monthLabels, weekdayShortLabels, firstDay }}
+        />
+      );
+    }
+
+    let clearButton;
+    if (this.isClearable()) {
+      clearButton = (
+        <i className="c-datepicker-clear icon-cross" onClick={this.handleClear} />
+      );
+    }
+
+    return (
+      <div ref={this.setEl('container')} className={className}>
+        <label onClick={this.handleClick}>
+          {labelNode}
+          <div className="c-datepicker-container">
+            <input
+              className={inputClassName}
+              placeholder={placeholder} value={localizedValue} readOnly
+            />
+            {clearButton}
+            <input ref={this.setEl('input')} type="hidden" value={value || ''} />
```

Comment:
we don't need two inputs, it was a hack to make pikaday work. matter of fact we don't need input elements at all. a simple span with ui-input will do just as well. you can't focus on this field properly anyway

Reply:
**Got rid of the hidden input.**

What's stopping me from switching over to span:
- `Datepicker extends InputComponent`: It has a getter called `getInput` which uses the input ref. It's not used anywhere but in my opinion it should be documented if input ref is required for InputComponent subclasses or not. I would expect every subclass of InputComponent to return a input element at `getInput()`

I think we should at least open up the picker when the field gets focused and hide it on blur. From a users perspective the field looks like a textfield, therefore it may cause confusion to either skip the field in tab-order or not show any indication of being focused.